### PR TITLE
Final PR

### PR DESCRIPTION
**Pull request by Alusp**
### We implemented the following in this PR:

- [ ] Follow Linters rule
- [ ] Follow Git flow rules
- [ ] Follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- [ ] Login page and registration page:
Should be built with Devise.

- [ ] Food list:

- Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
- Should lead to a form that allows users to add new food.

- [ ] Recipes list:

- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

- [ ] Public recipe list:

- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.

- [ ] Recipe details:

- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- [ ] General shopping list view (only for a group that has less than 3 members):

- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.

- [ ] Make sure there are no N+1 queries happening.

- [ ] Create a navigation menu that allows users to open all of the pages you created.

- [ ] Technical setup

- Set up the repository on GitHub and use Gitflow.
- Set up Devise for authentication.
- Set up RSpec and Capybara testing libraries.

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          </div>
+          <span class="ml-3 text-sm font-medium text-gray-400 dark:text-gray-500">private</span>
+        </div>
+      <% end %>
+    </div>
+  </div>
+
+  <div class="flex justify-around">
+    <% if @recipe.user == current_user %>
+      <%= link_to "Generate shoping list", general_shopping_list_path(@recipe.id), class: "border px-2 py-2 font-bold"%>
```
Comment (codepantha) [inline app/views/recipes/show.html.erb:35]: - [ ] When the "Generate shopping list" link is clicked, it is expected that: 
1. A page showing the list of food that is missing for all recipes of the logged-in user is rendered.
2. It should count the total food items and total price of the missing food.
However, none of that currently happens instead the following page is shown as you can see below 👇 

image


**This is what your general shopping list should look like** 👇
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Beautiful UI 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please write unit and integration tests with Rspec and Capybara as it is a project requirement
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (muneebulrehman) [reply app/views/recipes/show.html.erb:35]: Hey guys, once I click on the "Generate shopping list" it displays the public recipes page even though the URL path is for the shopping list page. Kindly fix this issue. You can refer to [this video](https://www.loom.com/share/6b6b8f9a0d8c469c98e5a2a0b6f36060) where I demonstrated the issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors
- Descriptive PR

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As requested by the previous reviewer, even though you implemented some tests, you still have a lot of tests to implement. There are almost no tests for controllers and also some tests for models are missing ( food, recipe_food).

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi Team, 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alusp** [comment]
Thanks very much for the review @sumairq

**Pull request merged**